tensorflow_datasets/image/stl10.py

```diff
@@ -0,0 +1,120 @@
+"""The STL-10 dataset is an image recognition dataset for developing \
+unsupervised feature learning, deep learning, self-taught learning algorithms. \
+It is inspired by the CIFAR-10 dataset but with some modifications.\
+ In particular, each class has fewer labeled training examples than in CIFAR-10,\
+ but a very large set of unlabeled examples is provided to learn \
+image models prior to supervised training. The primary challenge is to make use \
+of the unlabeled data (which comes from a similar but different distribution from the labeled data) to build a useful prior.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+import numpy as np
+import tensorflow_datasets.public_api as tfds
+
+
+_CITATION = """
+Adam Coates, Honglak Lee, Andrew Y. Ng An Analysis of Single Layer Networks in Unsupervised Feature Learning AISTATS, 2011.
+"""
+
+
+_DESCRIPTION = """
+The STL-10 dataset is an image recognition dataset for developing unsupervised
+feature learning, deep learning, self-taught learning algorithms. It is inspired
+by the CIFAR-10 dataset but with some modifications. In particular, each class
+has fewer labeled training examples than in CIFAR-10, but a very large set of
+unlabeled examples is provided to learn image models prior to supervised
+training.The primary challenge is to make use of the unlabeled data
+(which comes from a similar but different distribution from the labeled data) to build a useful prior.
+"""
+
+
+def read_all_images(images_path):
+    """
+    :param path_to_data: the file containing the binary images from the STL-10 dataset
+    :return: an array containing all the images
+    """
+    with open(images_path, 'rb') as f:
+        # tf.io.gfile.GFile doesn't works with np.fromfile as it doesn't has fileno() method
+        everything= np.fromfile(f,dtype=np.uint8)
+        images = np.reshape(everything, (-1, 3, 96, 96))
+        images = np.transpose(images, (0, 3, 2, 1))
+        return images
+
+
+def read_labels(path_to_labels):
+    """
+     :param path_to_labels: path to the binary file containing labels from the STL-10 dataset
+     :return: an array containing the labels
+    """
+    with tf.io.gfile.GFile(path_to_labels, 'rb') as f:
+        labels = f.read()
```

Comment:
f.read() returns a string, not an array.

Did you want to use f.readlines() ?
(you might also want to make sure that you strip those strings, to avoid "\n")